packages/mapper/src/mapper.ts

```diff
@@ -542,6 +553,40 @@ export default class Mapper {
     return result.value;
   }
 
+  async collection_renameCollection(
+    collection: Collection,
+    newName: string,
+    dropTarget?: boolean
+  ): Promise<any> {
+    if (typeof newName !== 'string') {
+      throw new Error('newName must be a string');
+    }
+
+    try {
+      await this.serviceProvider.renameCollection(
+        collection._database._name,
+        collection._name,
+        newName,
+        { dropTarget: !!dropTarget }
+      );
+
+      return {
+        ok: 1
+      };
+    } catch (e) {
+      if (e.name === 'MongoError') {
+        return {
+          ok: 0,
+          errmsg: e.errmsg,
```

Comment:
do you think it's possible to have this be just `message` as a standard error would have? This will not pass the `error` check in CLI's `writer` function, which explicitly checks for `message`

The other question I have, is whether you expect these to be handled differently than regular errors?

Reply:
Here i'm just re-creating the behaviour of the old shell. It looked so weird to me, i'd expect it to throw. 

We should not introduce breaking changes now, rather in a second moment and discuss it case by case, keep it on the back of your mind for now, we'll revisit this later, there is a chance it was done by mistake in the old shell.

Reply:
yea, that sounds good to me, we can reorganize it later.

The question that remains is how do you expect to handle `ok:1` and `ok:0` return types in the `writer`? Do we warn the user, do we tell them operation succeeded (ok: 1). Afaik, we have no other return types of this kind (and we will get more of them, I am sure). I just want to make sure we handle them properly. Happy if we do this in another PR though!

Reply:
Probably a shell api type would do that, but yep definitely another PR, we have plenty to do first, and for now the user is as good as it was in the old shell, so we are fine. 

I'd also bring this kind of topics to @max and go through prioritisation.